### PR TITLE
feat(query): Handle partial result response in RemoteExec

### PR DESCRIPTION
**Pull request by tjackpaul**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

### CODE REVIEW

File: query/src/main/scala/filodb/query/PromQueryResponse.scala [hunk 1]
```diff
@@ -8,14 +8,20 @@ final case class RemoteErrorResponse(status: String, errorType: String, error: S
...
 final case class ErrorResponse(errorType: String, error: String, status: String = "error") extends PromQueryResponse
 
-final case class SuccessResponse(data: Data, status: String = "success") extends PromQueryResponse
+final case class SuccessResponse(data: Data, status: String = "success",
+                                 isPartial: Option[Boolean] = None,
+                                 message: Option[String] = None) extends PromQueryResponse
 
-final case class ExplainPlanResponse(debugInfo: Seq[String], status: String = "success") extends PromQueryResponse
+final case class ExplainPlanResponse(debugInfo: Seq[String], status: String = "success",
+                                     isPartial: Option[Boolean]= None,
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/PromQueryResponse.scala:16]: Can explainPlan response be partial ?

File: query/src/main/scala/filodb/query/PromQueryResponse.scala [hunk 2]
```diff
@@ -8,14 +8,20 @@ final case class RemoteErrorResponse(status: String, errorType: String, error: S
 
 final case class ErrorResponse(errorType: String, error: String, status: String = "error") extends PromQueryResponse
 
-final case class SuccessResponse(data: Data, status: String = "success") extends PromQueryResponse
+final case class SuccessResponse(data: Data, status: String = "success",
+                                 isPartial: Option[Boolean] = None,
+                                 message: Option[String] = None) extends PromQueryResponse
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/PromQueryResponse.scala:13]: was the plan to use http response codes to indicate partial results, or attributes in json ?

**Review by vishramachandran — APPROVED** [review]
Approved with questions / comments.

File: query/src/main/scala/filodb/query/PromQueryResponse.scala (hunk 2 above)
Comment (tjackpaul) [reply query/src/main/scala/filodb/query/PromQueryResponse.scala:13]: Both. Http response code will be 206 and also, json will have a flag which can be used in grafana later if needed.

**Review by TanviBhavsar — APPROVED** [review]
If possible add a test case in PromCirceSupportSpec for partial respone

**Pull request merged**